### PR TITLE
Always send timings to statsd

### PR DESCRIPTION
**Pull request by gcatlin**
I noticed we only send timings to statsd on successful communication w/ Commitlog. This PR ensures we send timings to statsd on failed communication w/ Commitlog too.

### CODE REVIEW

File: src/postcommit_hook.erl [hunk 1]
```diff
@@ -28,26 +28,26 @@ attempt_send_to_kafka_riak_commitlog(Object, StartTime, RetryCount) ->
...
 
-    TimingResult = try
-        {Timing, ok} = timer:tc(?MODULE, sync_to_commitlog, [Action, Bucket, Key, Value]),
+    SyncStartTime = microtimestamp(),
+    try
+        sync_to_commitlog(Action, Bucket, Key, Value),
         error_logger:info_msg("[postcommit-hook] sync_to_commitlog success. Time: ~5.. B ms. Retries: ~p. ~s|~s",
-                              [timestamp()-StartTime, RetryCount, Bucket, Key]),
-        Timing
+                              [timestamp()-StartTime, RetryCount, Bucket, Key])
```
Comment (dtykocki) [inline src/postcommit_hook.erl]: Should this be `timestamp() - SyncStartTime`? Looks like the `StartTime` variable is unbound.

File: src/postcommit_hook.erl [hunk 2]
```diff
@@ -28,26 +28,26 @@ attempt_send_to_kafka_riak_commitlog(Object, StartTime, RetryCount) ->
...
     catch
         _:SyncException ->
             {SyncErrorReason, _} = SyncException,
             error_logger:info_msg("[postcommit-hook] Unable to sync data to Commitlog: ~p. Time: ~5.. B ms. Retries: ~p. ~s|~s",
-                                     [SyncErrorReason, timestamp()-StartTime, RetryCount, Bucket, Key]),
+                                  [SyncErrorReason, timestamp()-StartTime, RetryCount, Bucket, Key]),
```
Comment (dtykocki) [inline src/postcommit_hook.erl]: Same as above.

File: src/postcommit_hook.erl (hunk 1 above)
Comment (gcatlin) [reply src/postcommit_hook.erl]: No. But it is a little confusing.

`StartTime` is passed in to the fn on line 21. It provides millisecond granularity and represents the aggregate time for all sync attempts.

`SyncStartTime` is just for the current attempt.

Do you think renaming one or both vars would help?

File: src/postcommit_hook.erl (hunk 2 above)
Comment (gcatlin) [reply src/postcommit_hook.erl]: Same answer applies here.

File: src/postcommit_hook.erl (hunk 1 above)
Comment (gcatlin) [reply src/postcommit_hook.erl]: maybe `AggregateStartTime` and `AttemptStartTime`?

Comment (dtykocki) [reply src/postcommit_hook.erl]: Ahhh I see it now, the diff confused me a bit. The variable `SyncStartTime` naming isn't too confusing with the right context. It's pretty verbose but `SyncToCommitlogStartTime` makes it super clear IMO. It's your call here though.

**Comment by dtykocki** [comment]
>  I can do that as a separate PR. Any advice on a lightweight approach?

This could be opening a larger can of worms, but I'd start with plain old [EUnit](http://erlang.org/doc/apps/eunit/chapter.html) with sprinkles of [meck](https://github.com/eproxus/meck) for mocking out the external dependencies.

**Comment by gcatlin** [comment]
Closing this PR in favor of #23

**Pull request closed without merging**